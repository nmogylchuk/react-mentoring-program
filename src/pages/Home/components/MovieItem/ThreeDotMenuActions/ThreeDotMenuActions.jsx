import React, {useState} from 'react';
import MovieAddEdit from '../../ModalWindow/MovieAddEdit/MovieAddEdit';
import MovieDelete from '../../ModalWindow/MovieDelete/MovieDelete';
import CloseButton from '../../../../shared/CloseButton/CloseButton';
import {modals} from '../../../../../utils/constants';
import './ThreeDotMenuActions.scss';

const ThreeDotMenuActions = (props) => {
    const {closeMenu, movie} = props;

    const [modalType, setModalType] = useState(modals.DEFAULT);

    const showEditModal = () => {
        setModalType(modals.EDIT)
    }

    const showDeleteModal = () => {
        setModalType(modals.DELETE)
    }
    const closeModal = () => {
        setModalType(modals.DEFAULT)
    };

    return (
        <div>
            <div className="three-dot-menu-actions">
                <CloseButton closeMenu={closeMenu}/>
                <div className="three-dot-menu-actions__item" onClick={showEditModal}>
                    Edit Movie
                </div>
                <div className="three-dot-menu-actions__item" onClick={showDeleteModal}>
                    Delete Movie
                </div>
            </div>
            {modalType === modals.EDIT &&
            <MovieAddEdit onClose={closeModal} movie={movie} isEdit={true}/>}
            {modalType === modals.DELETE &&
            <MovieDelete closeModal={closeModal} id={movie.id}/>}
        </div>
    );
}

export default ThreeDotMenuActions;
