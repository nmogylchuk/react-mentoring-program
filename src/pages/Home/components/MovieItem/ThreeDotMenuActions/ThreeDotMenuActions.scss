@import '../../../../../assets/styles/variables';

.three-dot-menu-actions {
  position: absolute;
  right: 5px;
  top: 50px;
  display: flex;
  flex-direction: column;
  width: 180px;
  background-color: $dark-grey;

  &__item {
    cursor: pointer;
    min-width: 100%;
    padding: 10px 20px;
    margin: 0;

    &:hover {
      background-color: $red;
    }
  }
}
