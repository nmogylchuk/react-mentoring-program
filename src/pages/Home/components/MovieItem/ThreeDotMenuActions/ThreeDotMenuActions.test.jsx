import React from 'react';
import rerender from 'react-test-renderer';
import ThreeDotMenuActions from "./ThreeDotMenuActions";
import {movie} from '../../../../../../__mocks__/mockData';

describe('when Three Dot Menu Actions', () => {

    test('then snapshot created', () => {
        const threeDotMenuActionsComponent = rerender.create(<ThreeDotMenuActions movie={movie}/>);
        const tree = threeDotMenuActionsComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
