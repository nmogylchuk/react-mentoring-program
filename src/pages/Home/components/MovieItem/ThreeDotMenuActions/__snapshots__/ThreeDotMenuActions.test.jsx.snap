// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Three Dot Menu Actions then snapshot created 1`] = `
<div>
  <div
    className="three-dot-menu-actions"
  >
    <div
      className="close-button"
    />
    <div
      className="three-dot-menu-actions__item"
      onClick={[Function]}
    >
      Edit Movie
    </div>
    <div
      className="three-dot-menu-actions__item"
      onClick={[Function]}
    >
      Delete Movie
    </div>
  </div>
</div>
`;
